cores/arduino/stm32/usb/cdc/usbd_cdc_if.c

```diff
@@ -170,6 +168,9 @@ static int8_t USBD_CDC_Control(uint8_t cmd, uint8_t *pbuf, uint16_t length)
       linecoding.format     = pbuf[4];
       linecoding.paritytype = pbuf[5];
       linecoding.datatype   = pbuf[6];
+      if (linecoding.bitrate == 1200) {
+        jumpToBootloaderRequested();
+      }

```

Comment:
This resets directly when the baudrate is set to 1200. On the Arduino AVR implementation, the reset is triggered when the baudrate is 1200 and the port is *closed* (e.g. on `CDC_SET_CONTROL_LINE_STATE` with `(_usbLineInfo.lineState & 0x01) == 0`. See https://github.com/arduino/ArduinoCore-avr/blob/c8d6aef6d9331af1ec8dea6f78b4e43174cdb55d/cores/arduino/CDC.cpp#L97-L162 Additionally, the reset is not done immediately, but postponed for 120ms to allow the USB host to do additional cleanup. Either or both of these might be a good idea for the STM32 core too, I guess this could prevent serial port errors in the background caused by a reset while the port is still open.

The SAMD core does a similar thing, but with 250ms delay: https://github.com/arduino/ArduinoCore-samd/blob/2ca7f7531d4e4e54068bc584503b9e02e9033f3b/cores/arduino/USB/CDC.cpp#L155-L168

Reply:
Tanks for the links, I will check that to harden this feature.

Reply:
I found some more inspiration in the STM32L4 core from @GrumpyOldPizza, which also checks for *closing* the port at 1200 baud and starts a 250ms timeout: https://github.com/GrumpyOldPizza/arduino-STM32L4/blob/bacc184288a4644b2ee6a97672334983f3e788ab/system/STM32L4xx/Source/stm32l4_usbd_cdc.c#L222-L226

We should probably copy their approach to implement this timer using the USB SOF interrupt (which gives a nice way to delay reset, without having to reply on a timer or having to hook into the system tick timer: https://github.com/GrumpyOldPizza/arduino-STM32L4/blob/bacc184288a4644b2ee6a97672334983f3e788ab/system/STM32L4xx/Source/stm32l4_usbd_cdc.c#L66-L76

Reply:
Fine,
@GrumpyOldPizza, made a very good work on their and really optimized. ❤️

Reply:
The reason to use USB SOF is actually that for STM32L0 (and the next version of the STM32L4 code) we are using a tickless scheme where SysTick does not fire every 1ms. The thought was that if somebody is using USB power to begin with, ultra low power logic was not required.

Reply:
The reason to use USB SOF is actually that for STM32L0 (and the next version of the STM32L4 code) we are using a tickless scheme where SysTick does not fire every 1ms. The thought was that if somebody is using USB power to begin with, ultra low power logic was not required.

Reply:
The reason to use USB SOF is actually that for STM32L0 (and the next version of the STM32L4 code) we are using a tickless scheme where SysTick does not fire every 1ms. The thought was that if somebody is using USB power to begin with, ultra low power logic was not required.